### DISCUSSION TITLE
blink.cmp trigger words and how to configure.

### DISCUSSION THREAD

**Discussion by redoxahmii** [LazyVim/LazyVim#5155] [Q&A] [answered]
I know i should be asking this in `blink.cmp` but there isn't any replies there and I can't seem to figure out how trigger words are being handled in blink.
First was the issue with putting `{}` which i just saw has been fixed and was an upstream issue.
It still triggers for me whenever i enter insert after `>` or `,`
I did disable `opts.completions.trigger.show_on_trigger_character = false` and also tried adding these characters to `opts.completion.trigger.show_on_x_blocked_trigger_characters` but that does not seem to work at least for me.

If anyone figured this out help is appreciated!
Also waiting for `cmdline` cause the way i set it up it does not seem to trigger for me.

EDIT: seems like that upstream change has also made the trigger words work correctly. would appreciate the `cmdline` part still

**Comment by dpetka2001** [comment, marked as answer]
`opts.completions.trigger.show_on_trigger_character = false` seems to work for me on main branch. Probably not working on the release 0.7.6 and you should wait for the next release.

From my understanding `opts.completion.trigger.show_on_x_blocked_trigger_characters` is only for when accepting an entry or when entering insert mode. If you want to not show the completion menu when you are typing, then `trigger.show_on_trigger_character = false` or adding more trigger characters to block to `trigger.show_on_blocked_trigger_characters` table instead seems to be the way to go.

**Reply by redoxahmii** [reply]
Thank you for being helpful as always.
I was struggling most with the insert mode part and `opts.completion.trigger.show_on_x_blocked_trigger_characters` didn't seem to work for me yesterday and surprisingly it works now.
I did remedy for the time being with using `opts.completion.trigger.show_on_trigger_character = false` but as you said it disables it during typing as well and that wasn't my intended use case.
While i do find it triggering randomly with setting the above to `true` it seems a little less tedious than to configure `show_on_blocked_trigger_characters` table but i will look into it.
Now that the option for trigger words is working correctly it's been a little easier cause before it would trigger in the most random places.

**Reply by dpetka2001** [reply]
I personally don't have `show_on_trigger_character = false` but I do have 
```lua
show_on_accept_on_trigger_character = false,
show_on_insert_on_trigger_character = false,
```
Because I only like getting completions when I'm typing and not when entering `insert` mode or when accepting an entry. And I just added the `,` character to `show_on_blocked_trigger_characters` for now. Maybe will add some other stuff as things progress. 

I also saw that you got downvoted on Reddit for just mentioning that fzf-lua is faster and has a better API than telescope. Those reddit people looool

**Reply by redoxahmii** [reply]
They are just a different breed I usually just comment and don't reply afterwards lol

I will try your style as well cause `blink.cmp` tends to stutter for me on certain completions when typing and causes the content to be mistyped for me.
I thought i was losing it that i had forgotten how to type but i tested it again and again with the same word and it's the same behavior lol.
It stutters on a letter and the next inputs I provide during that stutter get typed before and than that letter is written lol

**Reply by redoxahmii** [reply]
For a showcase how it is happening :

https://github.com/user-attachments/assets/fdb07201-6800-4df2-8306-38214338226a


I tested with other words as well but this is the most consistent word I've found yet.
Do tell if this happens for you as well does not really happen on `nvim-cmp` for me.

**Reply by dpetka2001** [reply]
That's really weird. I've never experienced this before. But I only do Lua and Python. 

The only lag I've experienced is with `git-cmp` and `blink-compat` and ended up disabling the source in the end, because the letters I typed were slow to show up (like 0.5-1secs). This doesn't happen with `nvim-cmp`, so it's not a problem with `git-cmp` on its own. Seems like some kind of incompatibility in combination with `blink-compat`. I hope we will see more `blink` native sources in the future instead of relying in `blink-compat` to make `nvim-cmp` sources compatible with blink.

**Reply by redoxahmii** [reply]
I don't even know where this should be reported but i did test in Lua and Python and in plain `js` files but it is mostly present when in `jsx` or `tsx` .
I guess I will make an issue on `blink.cmp` for the time being.

I also tested a few sources with `blink.compat` initially and had sub par results which is why I kept using `nvim-cmp` for the time being but I honestly want to try switching to `blink.cmp` but there are still some issues like this that need time.

**Reply by dpetka2001** [reply]
Hopefully `blink.cmp` will grow up to mature. Even the maintainer of `nvim-cmp` has helped with some issues and he has admitted in the past that the design of `nvim-cmp` is not so flexible and instead of a rewrite it should be written from scratch, but of course he doesn't have the time to do that. 

The thing is that `nvim-cmp` has been around for so long and has become kind of a standard for plugins and completion sources. Hopefully, the scene for completion engines will become more diverse in the future, but yes it definitely needs some time to grow.

**Reply by redoxahmii** [reply]
Most people tend to forget the time period part that how long it has been since `nvim-cmp` and it's good to know that the author of `nvim-cmp` has expressed interest in `blink.cmp` taking its space and it matures to a much better product.
